File: producer/producer.js
import { Kafka, logLevel } from 'kafkajs'

const kafka = new Kafka({
    clientId: 'producer',
    brokers: ['localhost:9092'],
    logLevel: logLevel.WARN
})

const producer = kafka.producer()

let user = [
    {
        name: 'Vinicius Carvalho Miranda',
        email: 'vncs@gmail.com'
    }
]
console.log(user)
async function run() {
    await producer.connect()
    await producer.send({
        topic: 'topic-email',
        messages: [
            { value: JSON.stringify(user) },
        ],
    })
    await producer.disconnect()
}

run().catch(console.error)

File: Microservico-consumers/producer.py
from kafka import KafkaProducer
from kafka.errors import KafkaError
import json

producer = KafkaProducer(
    bootstrap_servers=['localhost:9092'],
    value_serializer=lambda m: json.dumps(m).encode('ascii')
    )

def on_send_success(record_metadata):
    print(record_metadata.topic)
    print(record_metadata.partition)
    print(record_metadata.offset)

def on_send_error(excp):
    log.error('I am an errback', exc_info=excp)

producer.send('pagamento', key=b'teste', value={
    'age': '26',
    'name':"Vinicius Miranda"
}).add_callback(on_send_success).add_errback(on_send_error)
producer.flush()

